perfkitbenchmarker/providers/digitalocean/digitalocean_disk.py

```diff
@@ -34,13 +61,71 @@ def Detach(self):
     pass
 
   def GetDevicePath(self):
-    # DigitalOcean VMs only have a single disk block device which is
-    # in use for the live filesystem, so it's not usable as a scratch
-    # disk device.
-    raise errors.Error('GetDevicePath not supported for DigitalOcean.')
+    # The local disk is always the boot disk, and it cannot be
+    # partitioned or reformatted, so we don't support GetDevicePath().
+    raise errors.Error(
+        'GetDevicePath not supported for DigitalOcean local disks.')
 
   def _Create(self):
     pass
 
   def _Delete(self):
     pass
+
+
+class DigitalOceanBlockStorageDisk(disk.BaseDisk):
+  """Interface to DigitalOcean Block Storage."""
+
+  def __init__(self, disk_spec, zone):
+    super(DigitalOceanBlockStorageDisk, self).__init__(disk_spec)
+    self.zone = zone
+    if self.disk_type != BLOCK_STORAGE:
+      raise ValueError('DigitalOcean data disks must have type block-storage.')
+    self.metadata = BLOCK_STORAGE_METADATA
+
+  def _Create(self):
+    self.volume_name = 'pkb-%s-%s' % (FLAGS.run_uri, self.disk_number)
+
+    response, retcode = util.DoctlAndParse(
+        ['compute', 'volume', 'create',
+         self.volume_name,
+         '--region', self.zone,
+         '--size', str(self.disk_size) + 'gb'])
+    if retcode:
+      raise errors.Resource.RetryableCreationError(
+          'Error creating disk: %s' % response)
```

Comment:
Might want to consider forcing the args into a tuple as you will get errors if it already one in this case, ie:

```
In [16]: fmt='hi %s'
In [17]: args='joe'
In [18]: fmt % args
Out[18]: 'hi joe'

In [20]: args='joe', 'smith'
In [21]: fmt % args
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-21-70797cd754bd> in <module>()
----> 1 fmt % args

TypeError: not all arguments converted during string formatting

In [22]: fmt % (args, )
Out[22]: "hi ('joe', 'smith')"
```